libvast/vast/system/disk_monitor.hpp

```diff
@@ -42,12 +48,20 @@ struct disk_monitor_state {
   /// Node handle of the INDEX.
   index_actor index;
 
-  /// The timespan between scans.
-  std::chrono::seconds scan_interval;
-
   constexpr static const char* name = "disk_monitor";
 };
 
+/// Contains values for the fields of `disk_monitor_state` with
+/// the same name.
+struct disk_monitor_config {
+  size_t high_water_mark;
+  size_t low_water_mark;
+  std::optional<std::string> scan_binary;
+  std::chrono::seconds scan_interval;
+};
+
+caf::expected<void> validate(const disk_monitor_config&);
```

Comment:
Please add documentation for this function. Also, this should probably return `caf::error` instead.